azure/resource-containers/pf-management-group/classes/pf-management-group-definition.ps1

```diff
@@ -1,12 +1,12 @@
-Class PfManagementGroupDefinition : PfDefinition{
-    [PfManagementGroupOptions]$Options
-    [PfDependency[]]$Dependencies
-    [AzSecurityDefinition]$Security
+Class PfManagementGroupDefinition : BaseAzResourceContainerDefinition{
+    PfManagementGroupDefinition() : base() {}
+    PfManagementGroupDefinition([PfBuildContext]$PfBuildContext) : base([PfBuildContext]$PfBuildContext){}
 
-    PfManagementGroupDefinition() {
-        $this.Options = (New-Object -TypeName "PfManagementGroupOptions")
+    [PfManagementGroupState]Deploy(){
+        $state = Deploy-PfManagementGroup -Definition $this
+        return $state
     }
-    PfManagementGroupDefinition([PfAzureContext]$PfAzureContext){
-        $this.Options = (New-Object -TypeName "PfManagementGroupOptions" -ArgumentList $PfAzureContext)
+    [void]Save(){
+        Save-PfManagementGroup -Definition $this
```

Comment:
I've now added this concept of "saving" your definition to the deployment context.  What this now allows us to do is decide whether or not the resource should actually be added to a deployment, and the greatest reason is it allows us to now run our process to find dependencies for that definition and prep it as needed for the deployment.  Could begin to build our "deployment plan" after each save if need be.  Let me know how you feel about this.